### PR TITLE
Iss hipm 722: Feature toggling

### PR DESCRIPTION
**Pull request by cbruegg**
## Description

This PR introduces the following changes:

- Implements `FeatureToggleApiAccess`
- Implements a `FeatureToggleRouter`:
  - Tries to get enabled features from server, fallback to default values defined in `FeatureConfiguration`
  - Can be updated during app runtime, updates feature state observers. Update is performed on app startup + login/logout. If the update fails, the previously used values remain in use.
- The achievements feature is enabled/disabled as part of this PR
- Remove unused `InverseBooleanConverter`

## Testing
To test this, you may change the value of the `Achievements` field in the `FeatureId` enum to `2`. The feature with id `2` currently does not exist, so the achievements feature should disappear. Changing it back to `1`, which is an enabled feature for all users, should make it appear again.

### CODE REVIEW

**Comment by cbruegg** [comment]
@Resudo This PR is ready to be reviewed now in full 😄

File: Sources/HiPMobile/BusinessLayer/ContentApiFetchers/ExhibitsBaseDataFetcher.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
```
Comment (Resudo) [inline Sources/HiPMobile/BusinessLayer/ContentApiFetchers/ExhibitsBaseDataFetcher.cs]: This looks like an unused using-statement. You should check if it is needed and possibly clean up the imports.

File: Sources/HiPMobile/BusinessLayer/FeatureToggling/FeatureToggleRouter.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        private readonly Dictionary<int, Observable<bool>> enabledFeatureObservables = new Dictionary<int, Observable<bool>>();
+
+        private FeatureToggleRouter(IList<int> enabledFeatureIds)
+        {
+            this.enabledFeatureIds = enabledFeatureIds;
+        }
+
+        public async Task RefreshEnabledFeaturesAsync()
+        {
+            var newEnabledFeatureIds = await FetchEnabledFeatureIdsAsync(fallbackEnabledFeatureIds: enabledFeatureIds);
```
Comment (Resudo) [inline Sources/HiPMobile/BusinessLayer/FeatureToggling/FeatureToggleRouter.cs]: The identifier `fallbackEnabledFeatureIds` seems unnecessary.

File: Sources/HiPMobile/BusinessLayer/UserManagement/UserManager.cs [hunk 3]
```diff
@@ -37,6 +38,7 @@ public class UserManager : IUserManager
                 Settings.Username = user.Username;
                 Settings.Password = user.Password;
                 Settings.AccessToken = user.Token.AccessToken;
+                await IoCManager.Resolve<IFeatureToggleRouter>().RefreshEnabledFeaturesAsync();
             }
 
             catch (Exception ex)
```
Comment (Resudo) [inline Sources/HiPMobile/BusinessLayer/UserManagement/UserManager.cs]: This is not part of your task, but if have the time, could you add brackets to the following lines? Thanks!

File: Sources/HiPMobile/ServiceAccessLayer/FeatureToggleApiDto/FeatureToggleApiAccess.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (C) 2017 History in Paderborn App - Universität Paderborn
```
Comment (Resudo) [inline Sources/HiPMobile/ServiceAccessLayer/FeatureToggleApiDto/FeatureToggleApiAccess.cs]: This file seems to be in the wrong folder. It should probably be in FeatureToggleApiAccess like the corresponding interface.

File: Sources/HipMobile.Tests/ServiceAccessLayer/ContentApiAccesses/FeatureToggleApiAccessTest.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        {
+            contentApiSubstitute = Substitute.For<IContentApiClient>();
+            client = new FeatureToggleApiAccess(contentApiSubstitute);
+        }
+
+        [Test, Category("UnitTest")]
+        public async Task GetEnabledFeaturesTest()
+        {
+            var json = @"
+[
```
Comment (Resudo) [inline Sources/HipMobile.Tests/ServiceAccessLayer/ContentApiAccesses/FeatureToggleApiAccessTest.cs]: Is this the way the test-json files have to be indented or can they be more aligned with the rest of the method?

File: Sources/HipMobileUI/ViewModels/Pages/MainPageViewModel.cs [hunk 6]
```diff
@@ -160,6 +199,8 @@ public override void OnDisappearing()
             base.OnDisappearing();
 
             SelectedViewModel.OnDisappearing();
+            achievementsFeatureSubscription?.Dispose();
+            achievementsFeatureSubscription = null;
```
Comment (Resudo) [inline Sources/HipMobileUI/ViewModels/Pages/MainPageViewModel.cs]: Is the explicit nulling of the `IDisposable` necessary?

**Pull request merged**